### PR TITLE
drag'n'drop improvements

### LINKED ISSUE #29063: A more intuitive way to open the group/tab properties in drag and drop form designer
Author Name: **Alexandre Heurion** (Alexandre Heurion)
Original Redmine Issue: [21245](https://issues.qgis.org/issues/21245)

Redmine category:forms

---
En utilisant un formulaire, nous voulons parfois qu'un groupe n'apparaisse que sous certaines conditions. Pour ça il existe une option "Configurer le Conteneur" qui s'ouvre en double-cliquant sur le sous-groupe. On peut alors y rentrer une expression contrôlant la visibilité du groupe (cf.image).
Cependant ce principe de double-clic n'est pas intuitif (comme un "clic droit > propriétés"). L'idéal serait (à mon avis) que les informations de cette boîte de dialogue apparaissent dans la colonne de droite (cf. image) comme informations de groupe.

Using the formularies, sometimes we wish a group only shows up with some conditions. There is an option for that (called in french "Configurer le Conteneur") as a dialog box that opens by double clickling on a group name. Then we can write some expression allowing the group to shows up or not (cf.image).
However this is not an intuitive way to use this option (not as much as a "right clic > properties" weel used with Windows). For me, it should be better if the options included in this box are displayed directly at the right of the screen (cf.image) like the informations of the field when we're selecting one.

---

- [formulaire Qgis champ optionnel 2.png](https://issues.qgis.org/attachments/download/14291/formulaire%20Qgis%20champ%20optionnel%202.png) (Alexandre Heurion)

### PR DESCRIPTION
**Pull request by 3nids**
part 3

this removes the widget config which was only accessible under double click
now a widget groupbox (by opposition to the field config) is shown on top of the right panel

second fix #29063

field:
image

relation:
image

last bits are the QML and HTML widget also available under double click
the double click is accessible from both trees but isn't saved if it's from the available widgets tree.
this would come as a part 4 hopefully.

### CODE REVIEW

**Comment by saberraz** [comment]
Nice work @3nids . The only thing missing is having a form preview :)

**Review by m-kuhn — APPROVED** [review]
:tada:

File: src/app/attributeformconfig/qgsattributewidgetedit.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/***************************************************************************
+    qgsattributewidgetedit.cpp
```
[the remark this answers is not in the document]
Comment (3nids) [reply src/app/attributeformconfig/qgsattributewidgetedit.cpp]: I'd leave it as since it's named accordingly to the others (like the relation edit)

**Comment by 3nids** [comment]
> The only thing missing now is having a form preview :)

@saberraz on the todo list :)

**Comment by obrix** [comment]
Hi @3nids , I am testing master right now and it seems I have some random crash I linked to your last commit : 

```
Fatal: ASSERT: "itemData.type() == QgsAttributesFormProperties::DnDTreeItemData::Container" in file /home/bertrand/Work/fork/QGIS/src/app/attributeformconfig/qgsattributewidgetedit.cpp, line 28
QGIS died on signal -1Could not attach to process.  If your uid matches the uid of the target
```

It happens when I click on fields on the available widget column or form layout column. Sometimes it crashes and sometimes it does not. Is something missing ?

**Comment by 3nids** [comment]
the assert is a left over, I am just checking why it's not crashing all the time...
I will fix it asap

**Comment by 3nids** [comment]
@obrix can you check if it's crashing only for containers (group box and tabs)?

**Comment by 3nids** [comment]
should be fixed in https://github.com/qgis/QGIS/pull/34503

**Comment by obrix** [comment]
I confirm its fixed. Thanks.

**Pull request merged**